common/cache/lru.go

```diff
@@ -157,10 +157,9 @@ func New(opts *Options) Cache {
 		rmFunc:        opts.RemovedFunc,
 		activelyEvict: opts.ActivelyEvict,
 		timeSource:    timeSource,
+		isSizeBased:   opts.IsSizeBased,
 	}
 
-	cache.isSizeBased = opts.GetCacheItemSizeFunc != nil && opts.MaxSize > 0
-
 	if cache.isSizeBased {
```

Comment:
let's add a log at the end of constructor to include whether it's size based and a few other useful information from the options such as initial capacity / max size etc.